scipy/spatial/transform/rotation.py

```diff
@@ -2091,24 +2117,26 @@ def __call__(self, times):
         if compute_times.ndim > 1:
             raise ValueError("`times` must be at most 1-dimensional.")
 
-        single_time = compute_times.ndim == 0
         compute_times = np.atleast_1d(compute_times)
 
-        # side = 'left' (default) excludes t_min.
-        ind = np.searchsorted(self.times, compute_times) - 1
-        # Include t_min. Without this step, index for t_min equals -1
-        ind[compute_times == self.times[0]] = 0
-        if np.any(np.logical_or(ind < 0, ind > len(self.rotations) - 1)):
+        ind = np.searchsorted(self.times, compute_times, side='right') - 1
+        if (np.min(compute_times) < np.min(self.times) or
+           np.max(compute_times) > np.max(self.times)):
             raise ValueError("Interpolation times must be within the range "
                              "[{}, {}], both inclusive.".format(
                                 self.times[0], self.times[-1]))
 
         alpha = (compute_times - self.times[ind]) / self.timedelta[ind]
 
-        result = (self.rotations[ind] *
-                  Rotation.from_rotvec(self.rotvecs[ind] * alpha[:, None]))
-
-        if single_time:
-            result = result[0]
-
-        return result
+        # Interpolate between rotations using a geometric slerp.
+        result = []
+        for i, a in zip(ind, alpha):
```

Comment:
This loop I believe can be vectorized quite easily.

Reply:
Fixed. Thanks.